test/plugins/sensor_demo/custom_sensor/CustomRenderingSensor.cc

```diff
@@ -0,0 +1,509 @@
+      << "a " << _sdf.TypeStr() << std::endl;
+  }
+
+  this->dataPtr->sdfSensor = _sdf;
+
+  if (this->Topic().empty())
+    this->SetTopic("/camera_left");
+  
+  this->dataPtr->pub_left =
+      this->dataPtr->node.Advertise<ignition::msgs::Image>("/camera_left");
+  this->dataPtr->pub_right =
+      this->dataPtr->node.Advertise<ignition::msgs::Image>("/camera_right");
+
+  if (!this->dataPtr->pub_left ||
+      !this->dataPtr->pub_right)
+  {
+    ignerr << "Unable to create publishers on topics" <<std::endl;
+    return false;
+  }
+
+  if (!this->AdvertiseInfo())
+    return false;
+
+  if (this->Scene())
+    this->CreateCamera();
+
+  this->dataPtr->sceneChangeConnection =
+      RenderingEvents::ConnectSceneChangeCallback(
+      std::bind(&CustomRenderingSensor::SetScene, this, std::placeholders::_1));
+
+  this->dataPtr->initialized = true;
+  return true;
+}
+
+//////////////////////////////////////////////////
+bool CustomRenderingSensor::Load(sdf::ElementPtr _sdf)
+{
+  sdf::Sensor sdfSensor;
+  sdfSensor.Load(_sdf);
+  return this->Load(sdfSensor);
+}
+
+/////////////////////////////////////////////////
+void CustomRenderingSensor::SetScene(ignition::rendering::ScenePtr _scene)
+{
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+  // APIs make it possible for the scene pointer to change
+  if (this->Scene() != _scene)
+  {
+    this->dataPtr->camera_left = nullptr;
+    this->dataPtr->camera_right = nullptr;
+    RenderingSensor::SetScene(_scene);
+    if (this->dataPtr->initialized)
+      this->CreateCamera();
+  }
+}
+
+//////////////////////////////////////////////////
+bool CustomRenderingSensor::Update(const std::chrono::steady_clock::duration &_now)
+{
+  IGN_PROFILE("CustomRenderingSensor::Update");
+  if (!this->dataPtr->initialized)
+  {
+    ignerr << "Not initialized, update ignored.\n";
+    return false;
+  }
+
+  if (!this->dataPtr->camera_left || !this->dataPtr->camera_right)
+  {
+    ignerr << "Camera doesn't exist.\n";
+    return false;
+  }
+
+  std::lock_guard<std::mutex> lock(this->dataPtr->mutex);
+
+  // move the camera to the current pose
+  this->dataPtr->camera_right->SetLocalPose(this->dataPtr->right_camera_pose);
+  this->dataPtr->camera_left->SetLocalPose(this->dataPtr->left_camera_pose);
+
+  this->dataPtr->generatingData = true;
+
+  // generate sensor data
+  this->Render();
+  {
+    IGN_PROFILE("CustomRenderingSensor::Update Copy image");
+    this->dataPtr->camera_left->Copy(this->dataPtr->image_left);
+    this->dataPtr->camera_right->Copy(this->dataPtr->image_right);
+  }
+
+  unsigned int width = this->dataPtr->camera_left->ImageWidth();
+  unsigned int height = this->dataPtr->camera_left->ImageHeight();
+  unsigned char *data_left = this->dataPtr->image_left.Data<unsigned char>();
+  unsigned char *data_right = this->dataPtr->image_right.Data<unsigned char>();
+
+  ignition::common::Image::PixelFormatType
+      format{common::Image::UNKNOWN_PIXEL_FORMAT};
```

Comment:
There's a compiler warning saying this variable (`format`) is set, but not used anywhere.

Reply:
removed the line